lib/datatype/index.js

```diff
@@ -108,4 +155,164 @@ export class DataType {
   decode(block) {
     return this.#decode(block)
   }
+
+  get key() {
+    return this.#writer.key
+  }
+
+  get keys() {
+    return this.cores.map((core) => {
+      return core.key.toString('hex')
+    })
+  }
+
+  get core() {
+    return this.#writer
+  }
+
+  get cores() {
+    return [...this.#corestore.cores.values()]
+  }
+
+  async info({ writerOnly = false } = {}) {
+    if (writerOnly) {
+      return this.#writer.info()
+    }
+
+    return Promise.all(
+      this.cores.map(async (core) => {
+        return core.info()
+      })
+    )
+  }
+
+  /**
+   * Get a doc by id
+   * @param {string} id
+   * @returns {Doc}
+   */
+  getById(id) {
+    return this.#sqlite.get(`select * from ${this.name} where id = '${id}'`)
+  }
+
+  /**
+   * Put a doc
+   * @param {Doc} data
+   * @returns {Promise<Doc>}
+   */
+  async put(data) {
+    await this.#writer.ready()
+
+    const created = data.created || new Date().getTime()
+    const doc = Object.assign({}, data, {
+      id: data.id || randomBytes(8).toString('hex'),
+      authorId: this.identityId,
+      version: `${this.#writer.key.toString('hex')}@${this.#writer.length}`,
```

Comment:
We would not actually store this on the doc written to the database, since it can be calculated on read (and putting it in the doc risks `this.#write.length` not being equal to the actual index when stored because of a race condition).